### DISCUSSION TITLE
how to show multiple modals?

### DISCUSSION THREAD

**Discussion by mohammad-zr** [mantinedev/mantine#3943] [Q&A] [answered]
First of all thanks for your great work. I know it's not best UX to show multiple modals but sometimes it's necessary. I want to show modals on top of other modals but nothing shown to me. Also tried setting z-index but it's not work. Can you help me to handle this issue? I really like this framework but also it's a big feature that I could not continue without that.

```
export const Modals = () => {
  const { modals, closeModal } = useModalStore();

  return (
    <>
      {modals?.map((modal: any, index) => {
        const { component: Component, props } = modal;

        return (
          <Modal.Root size={modal.size} key={index} opened onClose={() => closeModal()} centered>
            <Modal.Overlay />
            <Component {...props} />
          </Modal.Root>
        );
      })}
    </>
  );
};
```

**Comment by rtivital** [comment, marked as answer]
You can have multiple opened modals:

```ts
// first Modal
<Modal opened zIndex={400} />

// second modal
<Modal opened zIndex={500} />
```

**Comment by vaynevayne** [comment]
But you need to support drag and drop to make it convenient to use multiple modals, right?
